scripts/parse_specs.py

```diff
@@ -286,8 +286,8 @@ def __validate_members(d, tags):
             if not annotation:
                 raise Exception(prefix+"'desc' must start with {'[in]', '[out]', '[in,out]'}")
 
-            if type_traits.is_handle(item['type']):
-                raise Exception(prefix+"'type' must not be '*_handle_t': %s"%item['type'])
+            #if type_traits.is_handle(item['type']):
+            #    raise Exception(prefix+"'type' must not be '*_handle_t': %s"%item['type'])
```

Comment:
This is a workaround, need to investigate more why it's needed

Reply:
Worked out that because the loader adds a layer of indirection to handles, if we embed a handle in a struct this isn't done. To get this functionality to work I had to hack the following into the loader:
```
diff --git a/source/loader/ur_ldrddi.cpp b/source/loader/ur_ldrddi.cpp
index 4671709a..cb10d094 100644
--- a/source/loader/ur_ldrddi.cpp
+++ b/source/loader/ur_ldrddi.cpp
@@ -6885,6 +6885,19 @@ __urdlllocal ur_result_t UR_APICALL urCommandBufferUpdateKernelLaunchExp(
         reinterpret_cast<ur_exp_command_buffer_command_object_t *>(hCommand)
             ->handle;

+    uint32_t NumMemobjArgs = pUpdateKernelLaunch->numMemobjArgs;
+    if (NumMemobjArgs) {
+        for (uint32_t i = 0; i < NumMemobjArgs; i++) {
+            auto &MemobjArgDesc = pUpdateKernelLaunch->pArgMemobjList[i];
+            ur_mem_handle_t &ArgValue =
+                const_cast<ur_mem_handle_t &>(MemobjArgDesc.hArgValue);
+            ArgValue =
+                (ArgValue)
+                    ? reinterpret_cast<ur_mem_object_t *>(ArgValue)->handle
+                    : nullptr;
+        }
+    }
+
     // forward to device-platform
     result = pfnUpdateKernelLaunchExp(hCommand, pUpdateKernelLaunch);
```

Need to come up with a solution where this is generated by the scripts

Reply:
the changes in https://github.com/oneapi-src/unified-runtime/pull/1211 should allow this to be generated